resources/js/Components/Form/DmEntryCreateForm.tsx

```diff
@@ -165,33 +185,49 @@ const DmEntryCreateForm = ({
                     id='levels'
                     label='Levels'
                     name='Levels'
-                    type='number'
+                    InputLabelProps={{
+                        shrink: true,
+                    }}
                     InputProps={{
                         inputProps: {
                             min: 0,
+                            max: 20,
+                            inputMode: 'numeric',
+                            pattern: '[0-9]*',
                         },
                     }}
                     value={data.levels.toString()}
-                    onChange={(e) => setData('levels', parseInt(e.target.value))}
+                    onChange={(e) => {
+                        const value = parseInt(e.target.value)
+                        if (isNaN(value)) {
+                            setData('levels', 0)
+                        } else if (value < 0) {
+                            setData('levels', 0)
+                        } else if (value > 20) {
+                            setData('levels', 20)
+                        } else {
+                            setData('levels', value)
+                        }
+                    }}
                 />
                 {errors?.levels && <ErrorText message={errors?.levels} />}
             </StyledGrid>
             <StyledGrid item xs={12} md={type === 'Edit' ? 12 : 1}>
-                <TextField
+                <StyledGPTextField
                     fullWidth
                     id='gp'
                     label='GP'
                     name='GP'
                     type='number'
-                    InputProps={{
-                        inputProps: {
-                            min: 0,
-                        },
-                    }}
                     value={data.gp.toString()}
-                    onChange={(e) =>
-                        setData('gp', parseFloat(parseFloat(e.target.value).toFixed(2)))
-                    }
+                    onChange={(e) => {
+                        const value = parseFloat(parseFloat(e.target.value).toFixed(2))
```

Comment:
Can you explain why you have nested calls for the `parseFloat` function?

Reply:
I think @luigibesani implemented that so I don't know.